### PR TITLE
2 0 workflow transitions

### PR DESCRIPTION
**Pull request by ngilain**
[OO-FEATURE] Add workflow transitions form
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2064
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/720
https://github.com/open-orchestra/open-orchestra-model-interface/pull/278

### CODE REVIEW

File: Workflow/Form/DataTransformer/ProfileTransitionsTransformer.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+     * @param array $value
+     *
+     * @return array
+     */
+    public function transform($value)
+    {
+        $transitions = array();
+
+        foreach ($value as $transition) {
+            $transitions[] = $transition->getStatusFrom()->getId() . '-' . $transition->getStatusTo()->getId();
```
Comment (itkg-nanne) [inline Workflow/Form/DataTransformer/ProfileTransitionsTransformer.php]: you could put '-'  in a constant and factoryze his uses.

**Comment by itkg-nanne** [comment]
👍

File: Workflow/Factory/TransitionFactory.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+{
+    protected $transitionClass;
+
+    /**
+     * @param string $transitionClass
+     */
+    public function __construct($transitionClass)
+    {
+        if (!($transitionClass instanceof WorkflowTransitionInterface)) {
+            $transitionClass = WorkflowTransition::class;
```
Comment (alavieille) [inline Workflow/Factory/TransitionFactory.php]: throw a parameterExecption not use ModelBundle

File: Workflow/Form/DataTransformer/ProfileTransitionsTransformer.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    protected $statusRepository;
+    protected $transitionFactory;
+    protected $cachedStatuses = array();
+
+    /**
+     * @param StatusRepositoryInterface $statusRepository
+     * @param string                    $transitionClass
+     */
+    public function __construct(StatusRepositoryInterface $statusRepository, TransitionFactory $transitionFactory)
```
Comment (alavieille) [inline Workflow/Form/DataTransformer/ProfileTransitionsTransformer.php]: phpdoc

File: Workflow/Form/Type/Component/WorkflowTransitionsCollectionType.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults(
+            array(
+                'expanded' => 'true',
+                'multiple' => 'true',
+                'required' => false,
+                'statuses' => array(),
+                'locale'   => 'en'
```
Comment (alavieille) [inline Workflow/Form/Type/Component/WorkflowTransitionsCollectionType.php]: you can use parameter default locale ?

File: Workflow/Tests/Form/Type/Component/WorkflowProfileTransitionsTypeTest.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    /**
+     * Test getName
+     */
+    public function testGetName()
+    {
+        $this->assertSame('oo_workflow_profile_transitions', $this->form->getName());
+    }
+
+    public function testBuildForm()
```
Comment (alavieille) [inline Workflow/Tests/Form/Type/Component/WorkflowProfileTransitionsTypeTest.php]: phpdoc

File: WorkflowAdminBundle/Controller/Admin/TransitionController.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+     * @Config\Method({"GET", "POST"})
+     *
+     * @return Response
+     */
+    public function formAction(Request $request)
+    {
+        $this->denyAccessUnlessGranted(ContributionActionInterface::EDIT, WorkflowProfile::ENTITY_TYPE);
+        $currentLocale = $this->get('open_orchestra_backoffice.context_manager')->getCurrentLocale();
+        $profiles = $this->get('open_orchestra_model.repository.workflow_profile')
+            ->findBy(array(), array('labels.' . $currentLocale => 'asc'));
```
Comment (alavieille) [inline WorkflowAdminBundle/Controller/Admin/TransitionController.php]: you can use a repository method ?

File: Workflow/Form/DataTransformer/ProfileTransitionsTransformer.php [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * Generate a transition name from $statusFrom and $statusTo
+     *
+     * @param StatusInterface $statusFrom
+     * @param StatusInterface $statusTo
+     *
+     * @return string
+     */
+    public static function generateTransitionName(StatusInterface $statusFrom, StatusInterface $statusTo)
```
Comment (alavieille) [inline Workflow/Form/DataTransformer/ProfileTransitionsTransformer.php]: not use static

**Comment by ngilain** [comment]
@alavieille : update done

File: Workflow/Form/DataTransformer/ProfileTransitionsTransformer.php [hunk 8]
```diff
@@ -20,7 +20,7 @@ class ProfileTransitionsTransformer implements DataTransformerInterface
 
     /**
      * @param StatusRepositoryInterface $statusRepository
-     * @param string                    $transitionClass
+     * @param TransitionFactory         $transitionClass
```
Comment (alavieille) [inline Workflow/Form/DataTransformer/ProfileTransitionsTransformer.php]: phpdoc $transitionFactory

File: Workflow/Form/Type/Component/WorkflowProfileTransitionsType.php [hunk 9]
```diff
@@ -13,13 +13,16 @@
 class WorkflowProfileTransitionsType extends AbstractType
 {
     protected $dataClass;
+    protected $transitionTransformer;
 
     /**
-     * @param string $dataClass
+     * @param string                        $dataClass
+     * @apram ProfileTransitionsTransformer $transitionTransformer
```
Comment (alavieille) [inline Workflow/Form/Type/Component/WorkflowProfileTransitionsType.php]: @param

**Pull request merged**